packages/contracts/contracts/YieldGate.sol

```diff
@@ -93,23 +109,62 @@ contract BeneficiaryPool {
 
         gate = YieldGate(_gate);
         beneficiary = _beneficiary;
+
+        emitParametersChanged(0, 0);
+    }
+
+    // To save gas, add individual parameter setters.
+
+    function setMinAmount(uint256 _minAmount) external onlyBeneficiary {
+        minAmount = _minAmount;
+        emitParametersChanged(_minAmount, minDuration);
+    }
+
+    function setMinDuration(uint256 _minDuration) external onlyBeneficiary {
+        minDuration = _minDuration;
+        emitParametersChanged(minAmount, _minDuration);
+    }
+
+    function setParameters(uint256 _minAmount, uint256 _minDuration) external onlyBeneficiary {
+        minAmount = _minAmount;
+        minDuration = _minDuration;
+        emitParametersChanged(_minAmount, _minDuration);
+    }
+
+    function emitParametersChanged(uint256 _minAmount, uint256 _minDuration) internal {
+        emit ParametersChanged(beneficiary, _minAmount, _minDuration);
     }
 
     // Stakes the sent ether on behalf of the provided supporter. The supporter
     // is usually msg.sender if staking on the transaction sender's behalf.
+    // The staking timeout is reset on each call, so prior stake is re-locked.
     function stake(address supporter) public payable {
         uint256 amount = msg.value;
+        require(amount > 0 && amount >= minAmount, "amount too low");
```

Comment:
Shouldn't we instead somehow check whether `supporters[supporter] += amount >= minAmount`. Because this way it's not possible to top-up donations with less than `minAmount`, right?

Reply:
We can also do that. Just wanted to keep things super simple in this iteration. But I think this is a fair change, so I'll do it.